Empire/Empire.cpp

```diff
@@ -349,12 +350,17 @@ namespace {
             allocated_pp[group] += group_drawdown;  // relies on default initial mapped value of 0.0f
             group_pp_available -= group_drawdown;
 
-            float stockpile_drawdown = allocation <= group_drawdown ?  0.0f : (allocation - group_drawdown);
+            float stockpile_drawdown = 0.0f;
+            if (queue_element.item.build_type == BT_STOCKPILE) {
+                stockpile_transfer += group_drawdown;
```

Comment:
I think you also need to set ```allocation = group_drawdown``` here (or else adjust the initial calculation of allocation), or you will be churning the stockpile points (effectively delaying their availability for a turn and reducing the effectiveness of the stockpile build item).

Reply:
the initialisation to stockpile_drawdown to 0.0f is more of a leftover artefact here. I will move the calculations/cleanup that.

the turn delay is on purpose to keep it aligned what happens when there is excess (so you dont get micromanagement advantage out of the stockpiling project). Also I think its easier to understand the numbers for the user (probably that could be remedied by showing more in the UI). If we want that differently probably that should be discussed in the forum. For the moment this is the KISS thing, so i rather keep it like it is.

Reply:
I think you are misunderstanding my comment-- I am pointing out that it appears that you are currently letting the stockpile item itself draw from the stockpile if it does not have enough local group PP.  That undercuts the whole purpose of the stockpile item and seems wrong to me for multiple reasons: 1) can prevent other items from being able to use available stockpile due to using up the max usable per turn limit, 2) means that the stockpile item could actually wind up adding zero to the stockpile if its construction winds up being entirely funded from the stockpile, and 3) while whatever points do get assigned to it from the stockpile will get put back into the stockpile, the process of pulling them out and putting them back will tie them up for a turn.